src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,124 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { deleteComment, getComments, postComment } from '../../api/comments';
+import { TypePost } from '../../types';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+export const PostDetails = ({ post }) => {
+  const [comments, setComments] = useState([]);
+  const [isVisisble, setVisibility] = useState(true);
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+  useEffect(() => {
+    loadComments(post.id);
+  }, [post.id]);
 
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
+  const loadComments = (postId) => {
+    getComments(postId)
+      .then((commentsFromServer) => {
+        const newComments = commentsFromServer.filter((comment) => {
+          if (!comment.body.trim()) {
+            return;
+          }
 
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
+          comment.body.trim();
+          // eslint-disable-next-line
+          return comment;
+        });
+
+        setComments(newComments);
+      });
+  };
+
+  const addPostComment = (comment, postId) => {
+    postComment(comment, postId)
+      .then(() => {
+        loadComments(postId);
+      });
+  };
 
-        <li className="PostDetails__list-item">
+  const deletePostComment = (commentId, postId) => {
+    deleteComment(commentId)
+      .then(() => {
+        loadComments(postId);
+      });
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        <p>{post.body}</p>
+      </section>
+
+      <section className="PostDetails__comments">
+        {comments.length === 0 && 'There is no comment to display'}
+        {(comments.length > 0 && isVisisble === true) && (
+          <>
```

Comment:
```suggestion
```
No need for the fragment here. Everything is already in the 'section' tag

Reply:
Unfortunately, linter or even React itself requires a fragment  in this block  as well

Reply:
👍